### PR TITLE
Maintenance Cigarettes

### PR DESCRIPTION
**Pull request by thevandie**
## About The Pull Request

Adds three new types of cigarettes, as well as adds them to maintenance. Also adds nomads to maints because... why not, they're already in the contraband for the smoke vendor, and it's funny. Also, individual lucky strikes can spawn in trash piles, but it's kinda rare. 
They will all often be oldified, so players have to work around that.
image

The three new packs are:
Klassika Filter Cigarettes, Contains 3u Paracetemol, as a light painkiller. (Buckshot Roulette reference)
image
Kosmos Cigarettes, Contains 3u of Dexalin, because they're cosmonaut cigs, helps you breathe in space. (Quasimorph Reference)
image
Red Robusto Cigarettes, Contains 1u of Violence (Old SS13 Reference)
image

## Why It's Good For The Game

They add more flavor, more worldbuilding, and they each have unique effects, so it's fun. Plus maints was PREVIOUSLY devoid of cigarettes, so I decided to make a few old ones just for spice (instead of just having oldified space cigs, dromedaries, and acmes).

## Testing

Loaded up the game, tested all the oldification stuff, and all the cigs exist as items n shit, plus they spawn in piles like they're supposed to.

## Changelog
:cl:
add: Three new types of cigarettes for maintenance
tweak: These new cigarettes spawn in maints, inside of trash piles and occasional spawns. Nomads also spawn with them, old and moldy.
/:cl:

### CODE REVIEW

**Comment by Firefox13** [comment]
You know I think it would be REALLY cool if we had more "normal" cigarettes available in vendors instead of this powergamey shit.

There is no point hiding anything from people when they can see this PR.

**Comment by thevandie** [comment]
> You know I think it would be REALLY cool if we had more "normal" cigarettes available in vendors instead of this powergamey shit.

Well you see, they're in maints, not the vendor... plus, their chemical contents are so low it's basically just flavor. And I can't PUT them in the vendor because they were... y'know, specifically DESIGNED to be maintenance cigs, cigs from old companies that don't exist anymore. That was the whole point of this, to spice up maints.
And I didn't really just want them to have no gimmicks or differences, I thought that'd be a demerit on them existing. I mean, even the default cigs have stuff to differentiate them. Acmes have poison berry juice, nomads have adrenaline, etc.
As for the powergamer aspect...
the one unit of violence? It's one unit, it gives you a buff for like... 10 seconds I think? Very short, just for spice.
the three units of para? THREE units of a weak painkiller, barely does anything. Taking a swig of Vodka is a better idea.

so like I said, just for flavor, not powergamey at all honestly.

> There is no point hiding anything from people when they can see this PR.

I talked to people in shipyard about it, I'm removing that comment but just making the changelog shorter
cause I asked Hyperio and he said I didn't need to put EVERY minute detail in there :>

**Comment by Firefox13** [comment]
You need to chill out a little.
Did you really test the reagents? Are you sure 1u violence is only for 10 seconds, did you really test it?
> so like I said, just for flavor, not powergamey at all honestly.

Just for flavour would if they only contained only nicotine and sold in vendors (which I proposed in first comment. We're not repeating Nomads thing again), but this just boosts the cig smoking meta.

Yes, people aren't smart enough on their own to dip their own cigs in reagents for some reason already. Beats me.

Do your proposed ways of cleaning cigs work?

**Comment by thevandie** [comment]
> You need to chill out a little. Did you really test the reagents? Are you sure 1u violence is only for 10 seconds, did you _**really**_ test it?

Sorry, I have a 'tism and I write very formal educational replies for some reason, (I actually reduced it from how long it used to be).

And after testing it EXTENSIVELY, you do not get the buff at ALL. 1 violence, smoked, literally isn't enough to feel the effects. How 'bout that?
 
> Just for flavour would if they only contained only nicotine and sold in vendors (which I proposed in first comment. We're not repeating Nomads thing again), but this just boosts the cig smoking meta.

I'm not adding them to the vendors, because they exist to be maintenance items, which exists to make maints feel more mysterious/lived in. Acmes have poison berry juice, and dromedaries just have more nicotine. The idea was to have them be special in some way, even if it's minor. However, I'm perfectly fine with removing para from the klassikas, I just wanted them to have something that fit the reference. I am however NOT ok with removing the ONE singular unit of violence from the robustos, cause they're ROBUSTOS.

On a separate note, what mistake with nomads...? no one uses them for meta reasons, adrenaline does like... literally nothing. Unless it was something from like, months or years ago.

> Do your proposed ways of cleaning cigs work?

Yes. Yes it does. I have done it for basically a year now (every since i joined), and the dylo deletes the toxins right as they appear. I even tested it ingame specifically with the cigs, one toxin appears for like, half a second, then get nuked.

**Comment by Firefox13** [comment]
Or just put more Violence in it. lol.

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 1]
```diff
@@ -245,6 +247,30 @@
 	item_state = "luckpacket"
 	item_obj = /obj/item/clothing/mask/smokable/cigarette/lucky
 
+/obj/item/storage/fancy/cigarettes/robusto
+	name = "\improper Red Robusto packet"
+	desc = "A packet of six Red Robusto Cigarettes. Old NanoTrasen cigs, some say they make you more robust..."
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/fancy.dm:252]:
```suggestion
	desc = "A packet of six Red Robusto Cigarettes. Old Nanotrasen cigs, some say they make you more robust..."
```

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 2]
```diff
@@ -245,6 +247,30 @@
...
+	name = "\improper Klassika Filter packet"
+	desc = "A packet of six Klassika Filter Cigarettes. Old unknown brand, says 'Klassika Filter' on the front..."
+	icon_state = "bpacket"
+	item_state = "bpacket"
+	spawn_tags = SPAWN_TAG_CIGARETTE_MAINTS
+	item_obj = /obj/item/clothing/mask/smokable/cigarette/buckshot
+
+/obj/item/storage/fancy/cigarettes/quasimorph
+	name = "\improper Kosmos packet"
+	desc = "A packet of six Kosmos Cigarettes. Old Russian brand, says 'Kocmoc' on the front, along with a white rocket."
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/fancy.dm:268]:
```suggestion
	desc = "A packet of six Kosmos Cigarettes. Old Russian brand, says 'Космос' on the front, along with a white rocket."
```
Cyrillic is ought to work here

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 3]
```diff
@@ -245,6 +247,30 @@
...
+	icon_state = "kpacket"
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/fancy.dm:269]: I know you're just following existing sprite names but Jesus this is badly obfuscated

**Comment by Kurgis** [comment]
Only way I can see it working is if you take all the Violence cigs, put them into a mortar and pestle, and then form them into a either a drink or a pill. Since low chemical amounts don't do anything when smoking for some chems or at certain amounts. This sounds fine to me, not like crew provides tools to Vagabonds anyways.

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 1 above)
Comment (thevandie) [reply code/game/objects/items/weapons/storage/fancy.dm:252]: pretty sure the t is capitalized in most applications

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 3 above)
Comment (thevandie) [reply code/game/objects/items/weapons/storage/fancy.dm:269]: ah. fair. want me to rename it to kosmospacket and klassikapacket?

Comment (thevandie) [reply code/game/objects/items/weapons/storage/fancy.dm]: if yes, it's done. if not, unresolve.

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 1 above)
Comment (Firefox13) [reply code/game/objects/items/weapons/storage/fancy.dm:252]: It is both referred as NanoTrasen and Nanotrasen.
It should be Nanotrasen.

**Comment by thevandie** [comment]
So long as Firefox has no more qualms, this is done.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 4]
```diff
@@ -331,8 +333,39 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+	spawn_tags = SPAWN_TAG_CIGARETTE_MAINTS
+	type_butt = /obj/item/trash/cigbutt/buckshot
+
+/obj/item/clothing/mask/smokable/cigarette/quasimorph
+	desc = "A roll of tobacco and nicotine. Maint cig, ancient paper, how old is this...?"
+	preloaded_reagents = list("nicotine" = 6, "dexalin" = 3)
+	icon_state = "kosmosoff"
+	item_state = "kosmosoff"
+	icon_on = "kosmoson"
+	icon_off = "kosmossoff"
```
Comment (Firefox13) [inline code/game/objects/items/weapons/cigs_lighters.dm:365]:
```suggestion
	icon_off = "kosmosoff"
```
I only see `kosmosoff`, are you sure this sprite appears? Please double check.
Rest should be ok.

Comment (thevandie) [reply code/game/objects/items/weapons/cigs_lighters.dm:365]: image
yep

**Comment by thevandie** [comment]
just tested ingame, the commit works

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**